add.go

```diff
@@ -413,12 +446,46 @@ func (b *Builder) Add(destination string, extract bool, options AddAndCopyOption
 					return fmt.Errorf("invalid checksum flag: %w", err)
 				}
 			}
+
 			wg.Add(1)
-			go func() {
-				getErr = getURL(src, chownFiles, mountPoint, renameTarget, pipeWriter, chmodDirsFiles, srcDigest)
-				pipeWriter.Close()
-				wg.Done()
-			}()
+			if sourceIsGit(src) {
+				go func() {
+					var cloneDir string
+					cloneDir, _, getErr = define.TempDirForURL(tmpdir.GetTempDir(), "", src)
+
+					renamedItems := 0
+					writer := io.WriteCloser(pipeWriter)
+					writer = newTarFilterer(writer, func(hdr *tar.Header) (bool, bool, io.Reader) {
+						return false, false, nil
+					})
+					getOptions := copier.GetOptions{
+						UIDMap:         srcUIDMap,
+						GIDMap:         srcGIDMap,
+						Excludes:       options.Excludes,
+						ExpandArchives: extract,
+						ChownDirs:      chownDirs,
+						ChmodDirs:      chmodDirsFiles,
+						ChownFiles:     chownFiles,
+						ChmodFiles:     chmodDirsFiles,
+						StripSetuidBit: options.StripSetuidBit,
+						StripSetgidBit: options.StripSetgidBit,
+						StripStickyBit: options.StripStickyBit,
+					}
+					getErr = copier.Get(cloneDir, cloneDir, getOptions, []string{"."}, writer)
+					closeErr = writer.Close()
+					if renameTarget != "" && renamedItems > 1 {
```

Comment:
Line 456:
```
renamedItems := 0
```

Line 476:
```
if renameTarget != "" && renamedItems > 1 {
```

Is `renamedItems == 0` when evaluating the if statment in line 476? (It doesn't look like anything is modifying `renamedItems`)

Reply:
Thanks for catching this, fixed.